src/main/database_manager.cpp

```diff
@@ -87,6 +87,22 @@ const string &DatabaseManager::GetDefaultDatabase(ClientContext &context) {
 	return default_entry.catalog;
 }
 
+void DatabaseManager::SetDefaultDatabase(ClientContext &context, const string &new_value) {
+	optional_ptr<AttachedDatabase> db_entry;
+
+	context.RunFunctionInTransaction([&]() { db_entry = GetDatabase(context, new_value); });
```

Comment:
`RunFunctionInTransaction`  is very heavy-handed. It launches a new transaction if none is active (and commits it afterwards). Generally almost all code in DuckDB (be it parser, planner, optimizer, executor) is run with an active transaction already. Is this necessary only for the unittest? I imagine that when this function is used internally it will be used in locations where a transaction already exists.

Reply:
Indeed I added this to make a unit test case succeed in which I did not start a transaction before calling `SetDefaultDatabase`. Otherwise, an `InternalException` with message "TransactionContext::ActiveTransaction called without active transaction" would be thrown. I thought that it would be worth avoiding this exception by starting a new transaction, considering that `RunFunctionInTransaction` does not have a significant overhead when there is already an active transaction, does it?

Do you want me to remove `RunFunctionInTransaction` and just call `GetDatabase` directly?

Reply:
Yes, most functions in the catalog rely on a transaction existing. I think it is better to remove it.